lib/index.js

```diff
@@ -23,11 +23,11 @@ exports.clone = function (obj, seen) {
         return obj;
     }
 
-    seen = seen || { orig: [], copy: [] };
+    seen = seen || new Map();
 
-    const lookup = seen.orig.indexOf(obj);
-    if (lookup !== -1) {
-        return seen.copy[lookup];
+    const lookup = seen.has(obj);
```

Comment:
Could you do `seen.get(obj)` here and then just return that if it's truthy rather than checking for it first, and then doing the `.get`?